### DISCUSSION TITLE
rest in useEffect not work

### DISCUSSION THREAD

**Discussion by publicaccount365** [react-hook-form/react-hook-form#2832] [Q&A] [answered]
I wrote a component called InputV,
when fetch data from api, use reset for filled input,
input filled but ui does not work right.

https://codesandbox.io/s/empty-river-nx0b4?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
please use Controller for external controlled component: https://react-hook-form.com/api#Controller

**Reply by bluebill1049** [reply]
here is a working CSB: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r
